scripts/prog8.rb

```diff
@@ -0,0 +1,20 @@
+class Array
+    def power(x)
+        arr=[]
+        self.each do |value , temp=x|
+            final_val=1
+            while temp!=0
+                temp-=1
+                final_val*=value
+            end
+            arr<<final_val
+        end
+        print arr
+        puts
+    end
+end
+puts "Please provide an input"
+arr=$stdin.readline.split
```

Comment:
The input mentioned in the sheet includes double quotes(""). e.g. `"[-1,0,2,4,10]" 2`, but your program takes input without quotes. Please check this.